velocity/src/main/java/net/draycia/carbon/velocity/listeners/VelocityChatListener.java

```diff
@@ -75,78 +77,84 @@ private VelocityChatListener(
         this.carbonMessages = carbonMessages;
     }
 
-    @Subscribe(order = PostOrder.LATE)
-    public void onPlayerChat(final PlayerChatEvent event) {
-        if (!event.getResult().isAllowed()) {
-            return;
-        }
-
-        final Player player = event.getPlayer();
-        final boolean signedVersion = player.getIdentifiedKey() != null
-            && player.getProtocolVersion().compareTo(ProtocolVersion.MINECRAFT_1_19_1) >= 0;
-        if (signedVersion && !this.pluginManager.isLoaded("unsignedvelocity")) {
-            if (this.timesWarned.getAndIncrement() < 3) {
-                this.logger.warn("""
-                    
-                    ==================================================
-                    We have avoided modifying {}'s chat ,
-                    since they use a version higher than 1.19.1,
-                    where this function is not supported.
-                    
-                    If you want to keep this function working,
-                    install UnSignedVelocity.
-                    ==================================================
-                    """, player.getUsername()
-                );
+    @Override
+    public void register(final EventManager eventManager, final CarbonVelocityBootstrap bootstrap) {
+        eventManager.register(bootstrap, PlayerChatEvent.class, PostOrder.LAST, this);
+    }
+
+    @Override
+    public EventTask executeAsync(final PlayerChatEvent event) {
+        return EventTask.async(() -> {
```

Comment:
would prefer an extra method so we don't need to indent more

ie `return EventTask.async(() -> this.execute(event));`